fs_gcsfs/_gcsfs.py

```diff
@@ -186,14 +186,12 @@ def _scandir(self, path: str, return_info: bool = False, namespaces: List[str] =
         dir_key = self._path_to_dir_key(_path)
```

Comment:
maybe we should look into the `self._path_to_dir_key(_path)` why it does not find the right dir_key and prefix. The change here feels kind of random (e.g. have to change `dir_key = "/"` but `dfir_key = "/foo/` is fine